packages/server/lib/gui/auth.js

```diff
@@ -127,9 +127,12 @@ const _launchServer = (baseLoginUrl, sendMessage) => {
         res.redirect(`${baseLoginUrl}?status=${status}`)
       }
 
-      if (_.get(req.query, 'status') === 'error') {
-        authCallback(new Error('There was an error authenticating to the Cypress dashboard.'))
-        redirectToStatus('error')
+      if (_.get(req.query, 'status') === 'error' || !authCallback) {
+        if (authCallback) {
```

Comment:
it looks so weird, right: `if !authCallback` and then immediately `if (authCallback)`. Also you did not have `return redirectToStatus('error')` before - now you do (which is probably the right thing anyway)

Reply:
Yeah, I figured it was cleaner this way than two if statements that both call redirectToStatus, which is the alternative

Reply:
I think yeah, the final thing is the redirect, but for clarity to avoid tripping over next time I would make it more readable, something like this
```js
/**
 * nice little explanation of Electron < v5 and v5+ auth response
 */
const authFailed = (req, authCallback) => 
  // use dots - Lodash handles them and this protects against
  // undefined 'query' property inside 'req' object, also
  // has default value 'error' just in case 'req.query.status' is not set
  _.get(req, 'query.status', 'error') === 'error' || !authCallback
if (authFailed(req, authCallback)) {
  if (authCallback) {
    authCallback(...)
  }
  return redirect ...
}

Reply:
also @flotwig I am half-way done with writing a blog post showing how to replace these `if / if else` conditions with Maybe or Either functional structures that make code readable, just never had time to finish it

Reply:
Gotcha, but I think it's clearer in this case to have the conditions in-line. `!authCallback` doesn't mean that auth failed exactly, it's just there as a guard to prevent a crash in the case that `shell.openExternal` doesn't resolve after running or resolves immediately, since it's *technically* possible for `authCallback` to be undefined if Electron has a major bug. I inserted a comment so that this makes more sense.